### PR TITLE
Removing Moq as a dependency for mock unit tests

### PR DESCRIPTION
**Pull request by shacharPash**

### CODE REVIEW

File: tests/NRedisStack.Tests/TransactionsTests.cs [hunk 1]
```diff
@@ -9,8 +8,7 @@ namespace NRedisStack.Tests
 {
     public class TransactionTests : AbstractNRedisStackTest, IDisposable
     {
-        Mock<IDatabase> _mock = new Mock<IDatabase>();
-        private readonly string key = "TRX_TESTS";
+            private readonly string key = "TRX_TESTS";
```
Comment (haysch) [inline tests/NRedisStack.Tests/TransactionsTests.cs:11]: Seems like formatting went haywire here.

File: tests/NRedisStack.Tests/Json/JsonTests.cs [hunk 2]
```diff
@@ -103,10 +101,31 @@ public void TestSetFromDirectory()
     [Fact]
     public void TestJsonSetNotExist()
     {
+        //arrange
+        var conn = redisFixture.Redis;
+        var db = conn.GetDatabase();
+        IJsonCommands commands = new JsonCommands(db);
```
Comment (haysch) [inline tests/NRedisStack.Tests/Json/JsonTests.cs:107]: Inconsistent use of `var`/type declaration - intended?

File: tests/NRedisStack.Tests/Json/JsonTests.cs [hunk 3]
```diff
@@ -103,10 +101,31 @@ public void TestSetFromDirectory()
...
+        Assert.True(commands.Set("Person:Shachar", "$", obj, When.Exists));
+    }
+
+    [Fact]
+    public async Task TestJsonSetNotExistAsync()
+    {
+        //arrange
+        var conn = redisFixture.Redis;
+        var db = conn.GetDatabase();
+        IJsonCommandsAsync commands = new JsonCommands(db);
```
Comment (haysch) [inline tests/NRedisStack.Tests/Json/JsonTests.cs:122]: Inconsistent use of `var`/type declaration. Same as above.

**Review by haysch — COMMENTED** [review]
Just a minor inconsistency in variable declaration.

File: tests/NRedisStack.Tests/Json/JsonTests.cs (hunk 3 above)
Comment (shacharPash) [reply tests/NRedisStack.Tests/Json/JsonTests.cs:122]: If we use var then `commands` will be of type `JsonCommands` and not `IJsonCommands`.

Comment (haysch) [reply tests/NRedisStack.Tests/Json/JsonTests.cs:122]: What about using the `JsonCommandsAsync` constructor instead of casting?

Comment (shacharPash) [reply tests/NRedisStack.Tests/Json/JsonTests.cs]: I just tested locally and it works properly even when using var, I changed it in all the tests in `jsonTests.cs`.
Now it is consistent. What do you think?

Comment (haysch) [reply tests/NRedisStack.Tests/Json/JsonTests.cs]: Looks good!
As the implementation is (almost) based upon the same approach as `StackExchange.Redis`, where the `*Commands` inherits from `*CommandsAsync`, I do not have a strong opinion on using one or the other as long as the method matches the scenario 👍 

One could also use the extension methods to get the same result, i.e., `db.JSON()`.

**Review by chayim — APPROVED** [review]

**Pull request merged**